src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/SyntaxEditorBasedCodeFixProvider.FixAllProvider.cs

```diff
@@ -71,23 +71,23 @@ public sealed override async Task<CodeAction> GetFixAsync(FixAllContext fixAllCo
                     var updatedDocument = await task.ConfigureAwait(false);
                     currentSolution = currentSolution.WithDocumentSyntaxRoot(
                         updatedDocument.Id,
-                        await updatedDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false));
+                        await updatedDocument.GetSyntaxRootAsync(fixAllContext.CancellationToken).ConfigureAwait(false));
                 }
 
-                var title = fixAllState.GetDefaultFixAllTitle();
-                return new CodeAction.SolutionChangeAction(title, _ => Task.FromResult(currentSolution));
+                var title = FixAllContextHelper.GetDefaultFixAllTitle(fixAllContext);
+                return new CustomCodeActions.SolutionChangeAction(title, _ => Task.FromResult(currentSolution));
```

Comment:
❔ Since you aren't deriving from `SolutionChangeAction`, why not just call `CodeAction.Create`?

Reply:
Attempted to retain the existing semantics of this code.

Reply:
Tracked rationalizing use of CustomCodeActions in https://github.com/dotnet/roslyn/issues/41462